lib/tpm_hash.c

```diff
@@ -134,3 +136,75 @@ UINT32 tpm_hash_sequence(TSS2_SYS_CONTEXT *sapi_context, TPMI_ALG_HASH hash_alg,
                 (TPM2B_MAX_BUFFER *) &empty_buffer, TPM_RH_PLATFORM, result,
                 &validation, 0);
 }
+
+/* generate a unique app level error for fread failures */
+#define TSS2_APP_HASH_RC_FAILED (0x1 + 0x100 + TSS2_APP_ERROR_LEVEL)
+TPM_RC tpm_hash_files(TSS2_SYS_CONTEXT *sapi_context, file_list *inputs,
+        TPMI_ALG_HASH halg, TPMI_RH_HIERARCHY hierarchy, TPM2B_DIGEST *result, TPMT_TK_HASHCHECK *validation) {
+
+    TPM2B_AUTH nullAuth = TPM2B_EMPTY_INIT;
+    TPMI_DH_OBJECT sequenceHandle;
+
+    TPMS_AUTH_COMMAND cmdAuth = {
+        .sessionHandle = TPM_RS_PW,
+        .nonce = TPM2B_EMPTY_INIT,
+        .hmac = TPM2B_EMPTY_INIT,
+        .sessionAttributes = SESSION_ATTRIBUTES_INIT(0),
+    };
+    TPMS_AUTH_COMMAND *cmdSessionArray[1] = { &cmdAuth };
+    TSS2_SYS_CMD_AUTHS cmdAuthArray = { 1, &cmdSessionArray[0] };
+
```

Comment:
So we really want to concatenate all the files and hash them all, ie:
```
cat a b c | sha256sum
```
or do we want sha256sum a b c behavior?